src/merkle/smt/full/mod.rs

```diff
@@ -294,3 +299,62 @@ impl From<&RpoDigest> for LeafIndex<SMT_DEPTH> {
         Word::from(value).into()
     }
 }
+
+// SERIALIZATION
+// ================================================================================================
+
+impl Serializable for Smt {
+    fn write_into<W: ByteWriter>(&self, target: &mut W) {
+        let filled_leaves = self.entries().collect::<Vec<_>>();
```

Comment:
`entries()` returns an iterator. Should I not collect this in a Vec?

Reply:
I can use `self.entries().count()` instead. But in the background this also consumes the entire iterator. I changed it to that now

Reply:
> I can use self.entries().count() instead. But in the background this also consumes the entire iterator. I changed it to that now

Ah, I didn't fully realize the alternative, but given the code we have this may be the best option. Not great, but I hope the compiler does its thing. Maybe we should add a leaves counter later on.

Reply:
Thinking about it, if the trees get quite big, isn't it more efficient to collect them into a vector first, get the length of the vector, and then iterate over the vector?